### PR TITLE
Add cache busting for our files

### LINKED ISSUE #716: Add cache busting for our files

### PR DESCRIPTION
**Pull request by jmervine**
Fixes #716

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 'use strict';
 
-var sri = require('sri-toolbox');
-var fs  = require('fs');
+// create datestamp on startup
+const cacheBuster = Date.now().toString();
```
Comment (XhmikosR) [inline lib/helpers.js]: You should probably revert these to `var` since we don't use ES6 elsewhere.

**Comment by XhmikosR** [comment]
I don't like the query approach. I prefer the `filename.hash.ext` approach personally.

File: views/_partials/integrations.pug [hunk 2]
```diff
@@ -1,9 +1,11 @@
 hr
 br
 a(href=item.url, target="_blank")
-    img.img-responsive.thumbnail.center-block(src=item.img, alt=item.name)
+    != helpers.cacheBuster.img(item.img, { alt: item.name, class: "img-responsive thumbnail center-block" })
 h3
     a(href=item.url, target="_blank")=item.name
-| Platform:
```
Comment (XhmikosR) [inline views/_partials/integrations.pug]: The space is needed here.

File: views/_partials/showcase.pug [hunk 3]
```diff
@@ -1,10 +1,10 @@
 hr
 br
 a(href=item.url, target="_blank")
-    img.img-responsive.thumbnail.center-block(src=item.img, alt=item.name)
+    != helpers.cacheBuster.img(item.img, { alt: item.name, class: "img-responsive thumbnail center-block" })
 h3
     a(href=item.url, target="_blank")=item.name
-| Libraries:
```
Comment (XhmikosR) [inline views/_partials/showcase.pug]: And here.

File: lib/helpers.js [hunk 4]
```diff
@@ -40,14 +43,39 @@ function randomTweets(tweets, limit) {
...
+
+function jsWithBuster(uri, sri) {
+    return '<script src="' + uri + '?' + cachBuster +
+        '" integrity="' + sri + '" crissorigin="anonymous"';
+};
+
+function imgWithBuster(uri, attributes) {
+    var attrs = '';
+
+    for(var key in attributes) {
```
Comment (XhmikosR) [inline lib/helpers.js]: Missing space after `for`.

File: lib/helpers.js [hunk 5]
```diff
@@ -40,14 +43,39 @@ function randomTweets(tweets, limit) {
...
+        attrs += ' ' + key + '="' + attributes[key] + '"';
+    }
+
+    return '<img src="' + uri + '?' + cacheBuster + '"' + attrs + ' />';
```
Comment (XhmikosR) [inline lib/helpers.js]: No need to use `/>`. It's already self closing in HTML5, The same applies for `link`.

File: views/_partials/integrations.pug (hunk 2 above)
Comment (jmervine) [inline views/_partials/integrations.pug]: Right! Good catch, I have my EOL space removal set to ignore jade, need to update it to ignore pug too.

File: lib/helpers.js (hunk 1 above)
Comment (jmervine) [inline lib/helpers.js]: Okay, sure.

**Comment by XhmikosR** [comment]
On the other hand, my suggestion depends on #705, so I guess the query parameter solution is as good as it can get now.

**Comment by jmervine** [comment]
@XhmikosR I like that as well, and should we move to a precompiled static site setup, fronted by a CDN, I'd consider that. I think in the mean time though, this is a functional and acceptable solution. 

What I don't like is that it resets the cache buster on restart. I could write something in the startup that generates a sha of each file. I also considered using the sri for the css file.

**Comment by jmervine** [comment]
Addressed as suggested and rebased with `develop` so it should be right now.

**Comment by XhmikosR** [comment]
Please skip the Bootstrap file. I'll see later if it's something specific to nix.

On a side note, the tests fail.

**Comment by XhmikosR** [comment]
I just tested on Debian 8.5 and git 2.1.4 a fresh clone and works fine, i.e no modified files.

**Comment by jmervine** [comment]
I'm trying to skip it, I haven't been able to figure out how, even with the doc you sent. Do you want to checkout this branch and take a stab at it?

**Comment by XhmikosR** [comment]
I can try rebasing it skipping the offending file. Just note that you will
need to reset your branch after fetch.

**Comment by XhmikosR** [comment]
Done. Only thing is if we'll keep the whole tag in the JS or not.

**Comment by jmervine** [comment]
I'm not seeing your commit.

**Comment by XhmikosR** [comment]
There is no commit to see :p I just rebased against develop and skipped the
offending patch.

**Comment by jmervine** [comment]
RE the tag in the JS, I debated that. Since you mention it, it's probably better to just output the url with buster, then I don't need one function for each either. I'll make the change.

**Pull request merged**